### PR TITLE
microsoft.genpolicy: use git format-patch for patches

### PR DESCRIPTION
**Pull request by katexochen**

### CODE REVIEW

File: packages/by-name/microsoft/genpolicy/package.nix [hunk 1]
```diff
@@ -31,12 +31,16 @@ rustPlatform.buildRustPackage rec {
     patches = [
       # TODO(burgerdev): drop after Microsoft ported https://github.com/kata-containers/kata-containers/pull/9706
       (fetchpatch {
-        name = "genpolicy_device_support.patch";
+        name = "0001-genpolicy-add-rules-and-types-for-volumeDevices.patch";
```
Comment (burgerdev) [inline packages/by-name/microsoft/genpolicy/package.nix:34]: I'll take a note to vendor this patch, too.

**Review by burgerdev — APPROVED** [review]
Thanks!

File: packages/by-name/microsoft/genpolicy/package.nix (hunk 1 above)
Comment (katexochen) [reply packages/by-name/microsoft/genpolicy/package.nix:34]: If we want that I can easily do it now. I thought it wound't be necessary to vendor it.

Comment (katexochen) [reply packages/by-name/microsoft/genpolicy/package.nix:34]: Ah wait, the patch we are fetching isn't part of the upstream repo, got it.

**Pull request merged**